tools/serverpod_cli/lib/src/util/locate_modules.dart

```diff
@@ -3,83 +3,72 @@ import 'dart:io';
-        var moduleProjectRoot = List<String>.from(packageSrcRoot.pathSegments)
-          ..removeLast()
-          ..removeLast();
-        var generatorConfigSegments = path
-            .joinAll([...moduleProjectRoot, 'config', 'generator.yaml']).split(
-                path.separator);
-
-        var generatorConfigUri = packageSrcRoot.replace(
-          pathSegments: generatorConfigSegments,
-        );
-
-        var generatorConfigFile = File.fromUri(generatorConfigUri);
-        if (!await generatorConfigFile.exists()) {
-          continue;
-        }
-
-        var moduleProjectUri = packageSrcRoot.replace(
-          pathSegments: moduleProjectRoot,
-        );
-
-        var migrationVersions = findAllMigrationVersionsSync(
-          directory: Directory.fromUri(moduleProjectUri),
-          moduleName: moduleName,
-        );
-
-        var moduleInfo = loadConfigFile(generatorConfigFile);
-
-        var manualNickname = manualModules[moduleName];
-        var nickname = manualNickname ?? moduleInfo['nickname'] ?? moduleName;
-
-        modules.add(
-          ModuleConfig(
-            type: GeneratorConfig.getPackageType(moduleInfo),
-            name: moduleName,
-            nickname: nickname,
-            migrationVersions: migrationVersions,
-            serverPackageDirectoryPathParts: moduleProjectRoot,
-          ),
-        );
-      } catch (e) {
+  for (var packageInfo in packageConfig.packages) {
+    try {
+      var packageName = packageInfo.name;
+
+      if (!packageName.endsWith(_serverSuffix) && packageName != 'serverpod') {
+        continue;
+      }
+      var moduleName = moduleNameFromServerPackageName(packageName);
+
+      var packageSrcRoot = packageInfo.packageUriRoot;
+      var moduleProjectRoot = List<String>.from(packageSrcRoot.pathSegments)
+        ..removeLast()
+        ..removeLast();
+      var generatorConfigSegments = path
+          .joinAll([...moduleProjectRoot, 'config', 'generator.yaml']).split(
+              path.separator);
+
+      var generatorConfigUri = packageSrcRoot.replace(
+        pathSegments: generatorConfigSegments,
+      );
+
+      var generatorConfigFile = File.fromUri(generatorConfigUri);
+      if (!await generatorConfigFile.exists()) {
         continue;
       }
-    }
 
-    return modules;
-  } else {
-    log.error(
-      'Failed to read your server\'s package configuration. Have you run '
-      '`dart pub get` in your server directory?',
-    );
-    return null;
+      var moduleProjectUri = packageSrcRoot.replace(
+        pathSegments: moduleProjectRoot,
+      );
+
+      var migrationVersions = findAllMigrationVersionsSync(
+        directory: Directory.fromUri(moduleProjectUri),
+        moduleName: moduleName,
+      );
+
+      var moduleInfo = loadConfigFile(generatorConfigFile);
+
+      var manualNickname = manualModules[moduleName];
+      var nickname = manualNickname ?? moduleInfo['nickname'] ?? moduleName;
```

Comment:
Comment: Just realised that this can be a problem regarding the discussions we had yesterday on code generation and modules. Might even be a problem already? We are basically saying that the user project can override the nickname of a module. 

In the following scenario:
A -> B -> C
A -> C

Where A is your project, imagine that you give C the alias "carrot".
But module B has given C the alias "watermelon".

E.g. the graph looks like this:

A -> B-> watermelon (C).
A -> carrot (C).

I think we are okay right now, as we have duplication of the generated code, but this can be a pitfall when resolving modules in a more reasonable way.